posix: fs: ZFD_IOCTL_CLOSE: Be sure to call posix_fs_free_obj()

pfalcon opened the pull request:
To make sure that entry in fs.c:desc_array[] is freed. Note that
freeing an entry in fdtable is handled by generic implementation
of close().

Fixes: #17231

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

pfalcon committed:
posix: fs: ZFD_IOCTL_CLOSE: Be sure to call posix_fs_free_obj()

To make sure that entry in fs.c:desc_array[] is freed. Note that
freeing an entry in fdtable is handled by generic implementation
of close().

Fixes: #17231

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

aescolar reviewed (commented):
i don't see anything wrong with this, but I never dwell around this code.
@ramakrishnapallala you seem to have written most of this file, can you comment?

pfalcon committed:
tests: posix: test_fs_file: Cut down on noise output

Remove bunch of TC_PRINT's with too verbose logging of test operations.
The only logging required is in case of failures, and improve that in
a couple of cases by using zassert_*(), and log errno.

In particular, this allows to reuse existing testcase functions to
create more complex testcases.

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

pfalcon committed:
tests: posix: fs: Add test for (no) leak of file descriptors

Repeatedly open/close a file number of time larger than underlying
descriptor tables.

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

pfalcon commented:
A testcase for leaking descriptors is now added.
